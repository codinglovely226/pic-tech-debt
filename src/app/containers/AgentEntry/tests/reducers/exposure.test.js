/* eslint-disable */

import exposureReducer from '../../reducers/exposure';

import {
  addAgent,
  updateAgentDuration,
  updateAgentRouteOfExposure,
  updateAgentTimeOfExposure,
  updateAgentTimeSinceExposure,
  updateAgentTimeSinceExposureUnits,
  updateAgentTypeOfExposure,
  updateQuickViewAgentDuration,
  updateQuickViewAgentDurationUnits,
  updateQuickViewAgentDurationUnknown,
  updateQuickViewAgentRouteOfExposure,
  updateQuickViewAgentTimeOfExposure,
  updateQuickViewAgentTimeSinceExposure,
  updateQuickViewAgentTimeSinceExposureUnits,
  updateQuickViewAgentTypeOfExposure,
} from '../../actions/case';

describe('exposureReducer', () => {
  it('returns the initial state', () => {
    expect(exposureReducer(undefined, {})).toMatchSnapshot();
  });
  it('add agent', () => {
    expect(exposureReducer({}, addAgent(2, 'Test Agent', '1', 'mg', false, '2018-04-14 14:25:00', '1', 'hour', false, 'Test Exposure Type', 'Ingested', '123456'))).toMatchSnapshot();
  });
  it('update agent duration', () => {
    expect(exposureReducer({}, updateAgentDuration(1, 1, 1, '1', '2', 'hours', 'hours', false, false))).toMatchSnapshot();
  });
  it('update agent route of exposure', () => {
    expect(exposureReducer({}, updateAgentRouteOfExposure(1, 1, ['Old Exposure'], ['Old Exposure', 'New Exposure']))).toMatchSnapshot();
  });
  it('update agent time of exposure', () => {
    expect(exposureReducer({}, updateAgentTimeOfExposure(1, 1, 'Old Time of Exposure', 'New Time of Exposure'))).toMatchSnapshot();
  });
  it('update agent time since exposure', () => {
    expect(exposureReducer({}, updateAgentTimeSinceExposure(1, 1, 'Old Time Since', 'New Time Since'))).toMatchSnapshot();
  });
  it('update agent time since exposure units', () => {
    expect(exposureReducer({}, updateAgentTimeSinceExposureUnits(1, 1, 'Old Time Since Units', 'New Time Since Units'))).toMatchSnapshot();
  });
  it('update agent type of exposure', () => {
    expect(exposureReducer({}, updateAgentTypeOfExposure(1, 1, 'Old Type of Exposure', 'New Type of Exposure'))).toMatchSnapshot();
  });
  it('update quick view agent duration', () => {
    expect(exposureReducer({}, updateQuickViewAgentDuration(10))).toMatchSnapshot();
  });
  it('update quick view agent duration units', () => {
    expect(exposureReducer({}, updateQuickViewAgentDurationUnits('hours'))).toMatchSnapshot();
  });
  it('update quick view agent duration unknown', () => {
    expect(exposureReducer({}, updateQuickViewAgentDurationUnknown(true))).toMatchSnapshot();
  });
  it('update quick view agent route of exposure', () => {
    expect(exposureReducer({}, updateQuickViewAgentRouteOfExposure('Ingested'))).toMatchSnapshot();
  });
  it('update quick view agent time of exposure', () => {
    expect(exposureReducer({}, updateQuickViewAgentTimeOfExposure('Test TIme'))).toMatchSnapshot();
  });
  it('update quick view agent time since exposure', () => {
    expect(exposureReducer({}, updateQuickViewAgentTimeSinceExposure(10))).toMatchSnapshot();
  });
  it('update quick view agent time since exposure units', () => {
    expect(exposureReducer({}, updateQuickViewAgentTimeSinceExposureUnits('hours'))).toMatchSnapshot();
  });
  it('update quick view agent type of exposure', () => {
    expect(exposureReducer({}, updateQuickViewAgentTypeOfExposure('Acute'))).toMatchSnapshot();
  });
});
