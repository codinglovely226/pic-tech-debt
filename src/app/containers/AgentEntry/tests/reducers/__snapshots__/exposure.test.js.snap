exports[`exposureReducer add agent 1`] = `
Object {
  "duration": "1",
  "durationUnits": "hour",
  "durationUnknown": false,
  "route": "Ingested",
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "2018-04-14 14:25:00",
  "type": "Test Exposure Type",
}
`;

exports[`exposureReducer returns the initial state 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update agent duration 1`] = `
Object {
  "duration": "1",
  "durationUnits": "hours",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update agent route of exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [
    "Old Exposure",
    "New Exposure",
  ],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update agent time of exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "New Time of Exposure",
  "type": "",
}
`;

exports[`exposureReducer update agent time since exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "New Time Since",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update agent time since exposure units 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "New Time Since Units",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update agent type of exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "New Type of Exposure",
}
`;

exports[`exposureReducer update quick view agent duration 1`] = `
Object {
  "duration": 10,
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent duration units 1`] = `
Object {
  "duration": "",
  "durationUnits": "hours",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent duration unknown 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": true,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent route of exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": "Ingested",
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent time of exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "Test TIme",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent time since exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": 10,
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent time since exposure units 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "hours",
  "timestamp": "",
  "type": "",
}
`;

exports[`exposureReducer update quick view agent type of exposure 1`] = `
Object {
  "duration": "",
  "durationUnits": "",
  "durationUnknown": false,
  "route": Array [],
  "timeSince": "",
  "timeSinceUnits": "",
  "timestamp": "",
  "type": "Acute",
}
`;
